tools/runfiles/runfiles.rs

```diff
@@ -97,29 +99,34 @@ fn find_runfiles_dir() -> io::Result<PathBuf> {
             break;
         }
         // Follow symlinks and keep looking.
-        binary_path = binary_path.read_link()?;
-        if binary_path.is_relative() {
-            binary_path = env::current_dir()?.join(binary_path)
+        let link_target = binary_path.read_link()?;
+        binary_path = if link_target.is_absolute() {
+            link_target
+        } else {
+            let link_dir = binary_path.parent().unwrap();
+            env::current_dir()?.join(link_dir).join(link_target)
         }
     }
 
-    panic!("Failed to find .runfiles directory.");
+    Err(io::Error::new(
```

Comment:
I know this is tangential to the PR, so use your discretion on whether to address:

This is the first place I'll land if I'm a user doing a search in rules_rust for "Failed to find .runfiles". Does it make sense to add a comment above the line with a possible explanation or next steps?

Reply:
I think the explanation / next steps is 'you're doing some advanced bazel stuff and it went wrong'.. which is how we found it :)

I'm not confident enough in any explanation to add something right now.

Reply:
I am open to suggestion / including the snarky comment, lmk.

Reply:
Since this is the kind of "advanced bazel stuff", I think panicking is the correct way to go.

Reply:
I'm torn between "a library shouldn't panic" and "most users will unwrap the returned value"